tests/test_repositories.py

```diff
@@ -146,6 +146,29 @@ class p:
         ]
         assert mock.await_args[0][0][: len(expected_cmd)] == expected_cmd
 
+    async def test_cloning_with_custom_depth(self, monkeypatch):
+        """Ensure that we can retrieve the whole history, i.e. support true git clone"""  # noqa: E501
+
+        class p:
+            returncode = 0
+
+        mock = AsyncMock(return_value=p())
+        monkeypatch.setattr(prefect_gitlab.repositories, "run_process", mock)
+        repo = "git@gitlab.com:PrefectHQ/prefect.git"
+        depth = 0
```

Comment:
Once again, this suggestion is contingent on the suggestion of using `None` as the signal to fetch the entire git history.
```suggestion
        depth = None
```